rfc/monorepo.md

```diff
@@ -0,0 +1,557 @@
+
+No contract is given between numbered releases (`<year>.<number>`) - these may break, add, remove, or otherwise change any functionality deemed necessary by the project.
+
+#### Release Timing and Cadence
+
+Release timing and cadence is not addressed by this RFC.  When a new release is made during a calendar year, the release number increases by one.  When a patch release is made, the patch number increases by one.
+
+The release manager retains the authority to release as they see fit.  Backports to older versions will be judgements made based on the criticality of the issue, risk to a release's stability, and the difficulty of backporting the change itself.
+
+#### Release Branch Names
+
+Release branches will have a naming scheme similar to today, but date-oriented.  A release branch is prefixed by `release-`, appends the release version, and is suffixed with `.x`, e.g. `release-2020.3.x`.
+
+This branch will produce containers, libraries, and other artifacts for that numbered release.
+
+#### Artifact Versioning Scheme
+
+Spinnaker library and container versions from release branches will mirror the release version (e.g. `2023.3.4`).  This should be defined in the root `gradle.properties` file, and used throughout included builds and subprojects to publish their respective artifacts.
+
+From `main`, library and container versions will be published as `main-<gha_run_id>`, e.g. `main-4041166540`.  Github Actions run IDs always increase, so this functions identically to timestamp-based snapshot versions with two key benefits:
+
+1. The workflow that produced the artifact can be located trivially, which is extremely helpful when diagnosing many build and publish issues.
+    - A Docker container named `main-4041166540` or jar named `clouddriver-aws-main-4041166540` can be found at `https://github.com/spinnaker/spinnaker/actions/runs/4041166540`
+1. The shorter build ID is a lot more human readable, and there is no need for.
+
+This number follows [Gradle's rules for version ordering](https://docs.gradle.org/current/userguide/single_versions.html#version_ordering) to allow prefix patterns and version ranges to work naturally.  For example, the latest build of the `main` branch's Java libraries can be used by asking Gradle to find version `main-+`.
+
+Individual services/libraries will no longer have semver-shaped versions of their own.  Since all code is always built and published together, there is no need to declare any inter-service compatibility.  The compatibility contract is instead made against the project as a whole, rather than individual components.
+
+Examples:
+`main` branch publishes as: `<package>-main-<build_number>.jar`, e.g. `clouddriver-kubernetes-main-4041166540.jar`
+`release-<version>` branches publish as: `<java_package>-release-<full_version>.jar`, e.g. `clouddriver-kubernetes-release-2023.3.4.jar`
```

Comment:
MAIN concern with this is backports.  Given the above... is the plan for backports to go to "release-2022.3.6" from a "release-2022.3.5" version?  OR is it a "release-2022.4.0" situation?  When is a backport a minor vs. patch release?

Reply:
A backport would only be to a different numbered release (the second number, where the semver "minor" slot would be).  

> is the plan for backports to go to "release-2022.3.6" from a "release-2022.3.5" version?

`release-2022.3.6` would fully replace version `release-2022.3.5`, due to the non-breaking contract of a patch change described in the proposal.  There is no separate branch for these two versions - that scenario would be equivalent to `1.29.1` vs `1.29.2` today.

There is no concept of a "minor" release in this proposal - the suggested format is no contract between numbered releases (where the "minor" slot is), since that's a fuzzy judgement call that never seems to work out all that well in my experience.